server/api.go

```diff
@@ -317,15 +318,8 @@ func (p *Plugin) handlePostChannel(w http.ResponseWriter, r *http.Request, chann
 		}
 	}
 
-	data, err := ioutil.ReadAll(r.Body)
-	if err != nil {
-		res.Err = err.Error()
-		res.Code = http.StatusInternalServerError
-		return
-	}
-
 	var info ChannelState
-	if err := json.Unmarshal(data, &info); err != nil {
+	if err := json.NewDecoder(http.MaxBytesReader(w, r.Body, requestBodyMaxSizeBytes)).Decode(&info); err != nil {
```

Comment:
This is unrelated to the PR but something we should definitely fix so doing it here.